clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -5622,6 +5622,9 @@ void CodeGenModule::EmitGlobalVarDefinition(const VarDecl *D,
     getCUDARuntime().handleVarRegistration(D, *GV);
   }
 
+  if (LangOpts.HLSL)
+    getHLSLRuntime().handleGlobalVarDefinition(D, GV);
```

Comment:
Just checking you know that this has to work for non-globals as well? https://godbolt.org/z/MWzdTEb6q

Reply:
Interesting, is that really valid? When I try this with compute shader it generates `annotateHandle` but no `createHandle`: https://godbolt.org/z/Txr35hzac

In either case the binding is not listed in the Resource Bindings table and not included in metadata.

Reply:
Pre-SM6.6 this will emit an error (https://godbolt.org/z/e8nGf8cKf):
```
<source>:4:3: error: local resource not guaranteed to map to unique global resource.
```

Looks like a bug in DXC to me.

Reply:
Could someone file the bug please, so at least we have a record that it is a bug in DXC?

Reply:
Done: https://github.com/microsoft/DirectXShaderCompiler/issues/6971